components/hab/src/command/origin/key/download.rs

```diff
@@ -40,15 +40,16 @@ async fn handle_public(ui: &mut UI,
                        origin: &str,
                        revision: Option<&str>,
                        token: Option<&str>,
-                       cache: &Path)
+                       key_cache: &KeyCache)
                        -> Result<()> {
     match revision {
         Some(revision) => {
-            let nwr = format!("{}-{}", origin, revision);
-            ui.begin(format!("Downloading public origin key {}", &nwr))?;
-            match download_key(ui, api_client, &nwr, origin, revision, token, cache).await {
+            let named_revision = format!("{}-{}", origin, revision).parse()?;
```

Comment:
Yeah... I didn't do as much refactoring as I would have liked here. The API client really needs to be gone through to tighten up its interface with respect to these new types. There are a lot of places where we're taking name and revision of a key (and even returning those, I think). It could all be cleaned up, but it felt a bit out of scope for this PR.